File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/disk/adapter/items/HeaderItem.java
package com.semisonfire.cloudgallery.ui.main.disk.adapter.items;

public class HeaderItem extends DiskItem {

    private String date;
    private int count;

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    @Override
    public int getType() {
        return TYPE_HEADER;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof HeaderItem) {
            return date.equals(((HeaderItem)obj).getDate());
        }

        return super.equals(obj);
    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/disk/adapter/items/HeaderItem.kt
package com.semisonfire.cloudgallery.ui.disk.adapter.items

class HeaderItem : DiskItem() {

  var date: String? = null
  var count = 0

  override val type: Int
    get() = TYPE_HEADER

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (other !is HeaderItem) return false

    if (date != other.date) return false

    return true
  }

  override fun hashCode(): Int {
    return date?.hashCode() ?: 0
  }
}